kf_lib_data_ingest/common/file_retriever.py

```diff
@@ -162,6 +190,31 @@ def _file_save(protocol, source_loc, dest_obj, auth=None, logger=None):
         shutil.copyfileobj(orig, dest_obj)
 
 
+def _select_auth_scheme(url, auth_configs):
+    """
+    Select authentication scheme by inspecting `url`. If `url` starts with
+    any of the keys in auth_config, return the value, a dict containing
+    necessary parameters for the selected auth scheme, of the first key match.
+
+    See FileRetriever._validate_auth_config for details on auth_config format
+
+    :param url: the URL to be inspected
+    :type url: str
+    :param auth_configs: optional dict mapping URL patterns to authentication
+    schemes and necessary auth parameters
+    :type auth_configs: dict
+
+    :returns selected_cfg: configuration dict of the selected auth scheme
+    """
+    selected_cfg = None
+    if auth_configs:
+        for key, cfg in auth_configs.items():
+            if url.startswith(key):
+                selected_cfg = cfg
+                break
+    return selected_cfg
+
```

Comment:
I would like to advocate for removing the first and last two lines

```Python
    if auth_configs:
         for key, cfg in auth_configs.items():
             if url.startswith(key):
                 return cfg
```

Reply:
👍